internal/execution/worker.go

```diff
@@ -0,0 +1,84 @@
+package execution
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"time"
+
+	"github.com/riverqueue/river"
+
+	"github.com/zitadel/zitadel/internal/query"
+	exec_repo "github.com/zitadel/zitadel/internal/repository/execution"
+)
+
+type Worker struct {
+	river.WorkerDefaults[*exec_repo.Request]
+
+	config WorkerConfig
+	now    nowFunc
+}
+
+func (w *Worker) Timeout(*river.Job[*exec_repo.Request]) time.Duration {
+	return w.config.TransactionDuration
+}
+
+// Work implements [river.Worker].
+func (w *Worker) Work(ctx context.Context, job *river.Job[*exec_repo.Request]) error {
+	ctx = ContextWithExecuter(ctx, job.Args.Aggregate)
+
+	// if the event is too old, we can directly return as it will be removed anyway
+	if job.CreatedAt.Add(w.config.MaxTtl).Before(w.now()) {
+		return river.JobCancel(errors.New("event is too old"))
+	}
+
+	targets, err := TargetsFromRequest(job.Args)
+	if err != nil {
+		return river.JobCancel(errors.New("targets not valid"))
+	}
+
+	_, err = CallTargets(ctx, targets, exec_repo.ContextInfoFromRequest(job.Args))
+	if err != nil {
+		return river.JobCancel(err)
```

Comment:
JobCancel stops retries so I wouldn't cancel the job if a target wasn't available once.

Reply:
The `CallTargets` function includes all error handling for the list of targets, which means you only get an error returned if the `InterruptOnError` is set. That's why I think to cancel the Job is an indicator of the failed execution, or should that be handled differently?

Reply:
That's specifically how it is designed, there are no retries, and an error is only returned if the target is defined with `interruptOnError`. 
And with what do you disagree? I mean, should it be handled differently in your opinion? Do you disagree with any remark I made there? Would you handle the job differently?

Reply:
Yes I do disagree there, failure count does not necessarily mean that you need to retry, just that acknowledge that the call has failed. 
That you for example could handle if you create multiple users and every call failed, that you deactivate the execution.
Or are the failed queue jobs not visible anymore?

Reply:
ok please document this behavior so that it's clear for users of actions and update the acceptance criteria that we do not retry and therefore behave different than projections.